include/net/mqtt_socket.h

```diff
@@ -0,0 +1,655 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt.h
+ *
+ * @defgroup mqtt_socket MQTT Client library
+ * @ingroup networking
+ * @{
+ * @brief MQTT Client Implementation
+ *
+ * @details
+ * MQTT Client's Application interface is defined in this header.
+ *
+ * @note The implementation assumes TCP module is enabled.
+ *
+ * @note By default the implementation uses MQTT version 3.1.0.
+ *       However few cloud services like the Xively use the version 3.1.1.
+ *       For this please set client.protocol_version = MQTT_VERSION_3_1_1.
+ */
+
+#ifndef MQTT_H_
+#define MQTT_H_
+
+#include <zephyr/types.h>
```

Comment:
I recall the original Nordic SDK MQTT source used stdint.h types, which made the code a bit more portable.

Any reason we cannot continue to use stdint.h, as was done here:
e37c3b32a49b6e5ccba1da86ee8ceaa38f97c081 ?

Reply:
That's correct, source implementation used `stdint.h` types. I've changed them as I've thought Zephyr prefers to use `u8_t` etc, especially that checkpatch throws a warning when these standard types are used.
I have nothing against to `stdint.h` types though, so if we agree that it's fine to use it, I'll be happy to bring them back. @jukkar?